packages/swapkit/helpers/src/helpers/memo.ts

```diff
@@ -37,6 +51,29 @@ export const getMemoFor = <T extends MemoType>(memoType: T, options: MemoOptions
       return `${memoType}:${address}`;
     }
 
+    case MemoType.SWAP: {
+      const {
+        asset,
+        address,
+        limit,
+        ssInterval,
+        ssQuantity,
+        affiliateAddress,
+        affiliateFee,
+        dexAggAddress,
+        dexAggAsset,
+      } = options as MemoOptions<MemoType.SWAP>;
+
+      const limitWithSS = limit
+        ? `${limit}/${ssInterval ? `${ssInterval}` : ""}/${ssQuantity ? `${ssQuantity}` : ""}`
+        : "";
+      return `${asset?.toString()}:${address}:${limitWithSS}:${
+        affiliateAddress ? `${affiliateAddress}` : ""
+      }:${affiliateFee ? `${affiliateFee}` : ""}:${dexAggAddress ? `${dexAggAddress}` : ""}:${
+        dexAggAsset ? `${dexAggAsset}` : ""
+      }`;
```

Comment:
That part is definitely `too complex` to be binded as those ternaries. Use more variables and be more descriptive or extract those to functions. by having multiple ternaries - code become unreadable.